packages/field-base/src/field-mixin.js

```diff
@@ -103,49 +99,106 @@ export const FieldMixin = (superclass) =>
     ready() {
       super.ready();
 
-      if (this._errorNode) {
-        this._errorNode.id = this._errorId;
+      const error = this._errorNode;
+      if (error) {
+        error.id = this._errorId;
 
-        this._applyCustomError();
+        this.__applyCustomError();
 
         this._updateErrorMessage(this.invalid, this.errorMessage);
       }
 
-      if (this._helperNode) {
-        this._currentHelper = this._helperNode;
-        this._helperNode.id = this._helperId;
-
-        this._applyCustomHelper();
-
-        this.__helperSlot = this.shadowRoot.querySelector('[name="helper"]');
-        this.__helperSlot.addEventListener('slotchange', this.__onHelperSlotChange.bind(this));
-        this.__helperIdObserver = new MutationObserver((mutationRecord) => {
-          mutationRecord.forEach((mutation) => {
-            // only handle helper nodes
-            if (
-              mutation.type === 'attributes' &&
-              mutation.attributeName === 'id' &&
-              mutation.target === this._currentHelper &&
-              mutation.target.id !== this.__savedHelperId
-            ) {
-              this.__updateHelperId(mutation.target);
-            }
+      const helper = this._helperNode;
+      if (helper) {
+        this.__applyCustomHelper(helper);
+      }
+
+      this.__helperSlot = this.shadowRoot.querySelector('[name="helper"]');
+
+      this._observer = new FlattenedNodesObserver(this.__helperSlot, (info) => {
+        const helper = this._currentHelper;
+
+        const newHelper = info.addedNodes.find((node) => node !== helper);
+        const oldHelper = info.removedNodes.find((node) => node === helper);
+
+        if (newHelper) {
+          // Custom helper is added, remove the default one.
```

Comment:
Assuming this branch gets selected in case the user adds one more custom helper node, I would suggest changing the comment text a bit in a way like:

```suggestion
          // Custom helper is added, remove the previous one.
```